core/src/main/java/me/aurium/beetle/branch/block/CommonBlockPath.java

```diff
@@ -0,0 +1,178 @@
+package me.aurium.beetle.branch.block;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class CommonBlockPath implements BlockPath {
+
+    public final static String COMMON_SPLITTER = "-";
+
+    private final BlockPath root;
+    private final BlockPath parent;
+    private final LinkedList<Block> blocks;
```

Comment:
You likely don't want to use a LinkedList. ArrayList is better in the vast majority of scenarios

Reply:
Linked list is better for inserting, right? Elements are to be inserted and removed constantly with paths with less reads

Reply:
I don't see any case in this class where a LinkedList would out-perform an ArrayList.

Unless you are absolutely sure you need a LinkedList, use an ArrayList. ArrayList is better for iterating and better for adding and inserting to the back in all scenarios. ArrayList is still better for inserting at the front and removing at the front than LinkedList, except when the List is very large.

In the very rare case that LinkedList does indeed out-perform ArrayList, look at ArrayDeque before considering LinkedList. ArrayDeque will likely beat whatever usage of LinkedList makes it better than ArrayList.

In the very, very rare case that you indeed do need a LinkedList, then use it.

Reply:
I have no idea what an ArrayDeque is (never needed one before) haha

I'll probably replace it with an ArrayList, but the reason i was using linked lists is the native support to add front and add back, which i don't think arraylist or List supplies.

Reply:
Resolved